vms/platformvm/state/diff.go

```diff
@@ -25,10 +28,12 @@ var (
 type Diff interface {
 	Chain
 
+	SetBlockID(ids.ID)
```

Comment:
This is an edge I didn't foresee. Currently each Diff dont know the ID of the block they referece.
I think the reason is mempool: we use a diff to verify mempool txs but we don't have yet a block for that diff.

I think a better solution than this horrible SetBlockID would be to pass blockID via the Diff constructor, pass ids.EmptyID for the mempool diffs and also err is blockID is empty if MerkleView is ever called on a mempool diff.